src/types/validate.go

```diff
@@ -252,6 +255,18 @@ func (action ZarfComponentAction) Validate() error {
 	return err
 }
 
+// ValidateReleaseName validates against DNS 1035 kubernetes label spec
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names
+func ValidateReleaseName(releaseName string) error {
```

Comment:
Does this need to be exported?

Reply:
It probably doesn't. I was halfway thinking it didn't need to be a method at all be erred on the side of being more explicit. I'll make it private (unless you have opinions about it being just part of the `Validate` method instead since it's simple conditionals.

Reply:
IMO just include this in the `func (chart ZarfChart) Validate()` method

Reply:
better yet just use the helm [ValidateReleaseName function](https://github.com/helm/helm/blob/a958f3a9b65df7c809c88c180cc89886fee7768c/pkg/chartutil/validate_name.go#L79)

Reply:
@AustinAbro321 Thanks for your suggestion! I actually decided not to use the Helm validation method due to how Kubernetes utilizes the releaseName later on. The distinction is subtle but important:

1. Helm's regex allows for periods `.` in the release name.
2. However, Kubernetes applies stricter rules for certain resources (like Services) using the DNS-1035 label standard.
3. This stricter standard doesn't allow periods and requires names to start with a letter. 

That being said, after considering this further, we could use what's in the [Kubernetes API](https://github.com/kubernetes/kubernetes/blob/974f3d3d8ff6bfb33a375f7207c34c69b3e8b932/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L235-L244) if it's preferred?

To summarize, my understanding is that the constraints in Helm are more lenient than in Kubernetes. Because Zarf abstracts away some of the configuration (a service receiving the releaseName in this case), I think it would make sense for us to be more strict in the Zarf package to prevent potential issues downstream.

What do you think about this approach? I'm open to discussing further if you have any concerns or alternative suggestions.

Reply:
It was really easy to add this, so I did. But in order to maintain the original way tests functioned, while also accounting for the error string that the k8s api yields (which is rather length) it added a small amount of conditionals to the existing tests. If that is a problem let me know, and I'll be happy to tweak further.